src/test/java/com/google/cloud/genomics/utils/grpc/FaultyGenomicsServerITCase.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.genomics.utils.grpc;
+
+import io.grpc.ManagedChannel;
+import io.grpc.Server;
+import io.grpc.Status;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.stub.StreamObserver;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Random;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.cloud.genomics.utils.IntegrationTestHelper;
+import com.google.cloud.genomics.utils.ShardBoundary;
+import com.google.cloud.genomics.utils.ShardUtils;
+import com.google.common.collect.ImmutableList;
+import com.google.genomics.v1.StreamVariantsRequest;
+import com.google.genomics.v1.StreamVariantsResponse;
+import com.google.genomics.v1.StreamingVariantServiceGrpc;
+
+@RunWith(JUnit4.class)
+public class FaultyGenomicsServerITCase {
+  public static final String SERVER_NAME = "integrationTest";
+  
+  protected static Server server;
+  protected static IntegrationTestHelper helper;
+  protected static GenomicsChannel genomicsChannel;
+  protected static double faultPercentage = 0.0;
+
+  /**
+   * Starts the in-process server that calls the real service.
+   * 
+   * @throws GeneralSecurityException
+   * @throws IOException
+   */
+  @BeforeClass
+  public static void startServer() throws IOException, GeneralSecurityException {
+    try {
+      server = InProcessServerBuilder.forName(SERVER_NAME)
+              .addService(StreamingVariantServiceGrpc.bindService(new VariantsIntegrationServerImpl()))
+              .build().start();
+    } catch (IOException ex) {
+      throw new RuntimeException(ex);
+    }
+    helper = new IntegrationTestHelper();
+    genomicsChannel = GenomicsChannel.fromOfflineAuth(helper.getAuth());
+  }
+
+  @AfterClass
+  public static void stopServer() {
+    server.shutdownNow();
+  }
+
+  protected static class VariantsIntegrationServerImpl implements
+      StreamingVariantServiceGrpc.StreamingVariantService {
+    final Random random = new Random();
```

Comment:
Travis CI is currently running the unit tests but not the integration tests.  The integration tests are run as part of the release process and also on demand.

On a separate note, I had a flaky unit test because I was not adequately protecting the variables controlling the retry behavior from the channel and service threads interacting with them.   With that fix, I think this PR is now good to go, but please let me know if there are any other changes you would like to see in this PR?  Thanks!